includes/class-utilities.php

```diff
@@ -11,17 +11,14 @@
  * The Class for Admin Settings Page and functions to access Settings Values.
  */
 class Utilities {
-
 	/**
-	 * Get the top level domain name from the site URL.
+	 * Get the domain name from the site URL.
 	 *
-	 * @return string the top level domain name.
+	 * @return string the domain name.
 	 */
-	public static function get_top_level_domain() {
-		$site_url     = get_site_url();
-		$domain_name  = wp_parse_url( $site_url, PHP_URL_HOST );
-		$domain_parts = explode( '.', $domain_name );
-		return sanitize_text_field( implode( '.', array_slice( $domain_parts, -2 ) ) );
+	public static function get_site_domain() {
+		$site_url = get_site_url();
```

Comment:
In multisite, plugins can only be installed on the network level. Therefore, the URL for the API should be the main site's URL.

With a multisite setup of:
- **First site: https://my-site.com** (main site)
- **Second site: https://my-other-site.com**

When on **Second site**, `get_site_url()` will return `https://my-other-site.com`.

Use `network_site_url()` instead, as it will return `https://my-site.com`.

Reply:
Agree with above